basxconnect/core/layouts/relationshipstab.py

```diff
@@ -0,0 +1,159 @@
+                primary_button=button_add_relationship_from(modal_from),
+            ),
+            modal_from,
+        ),
+    )
+
+
+def relationships_datatable(request, queryset, primary_button, title):
+    return layout.datatable.DataTable.from_model(
+        Relationship,
+        queryset,
+        title=title,
+        backurl=request.get_full_path(),
+        prevent_automatic_sortingnames=True,
+        columns=[
+            person_in_relationship(
+                "Person A",
+                "person_a",
+                lambda relationship: relationship.person_a,
+            ),
+            "type",
+            person_in_relationship(
+                "Person B",
+                "person_b",
+                lambda relationship: relationship.person_b,
+            ),
+            "start_date",
+            "end_date",
+        ],
+        rowactions=[
+            Link(
+                href=ModelHref(
+                    Relationship,
+                    "edit",
+                    kwargs={"pk": hg.C("row.pk")},
+                    query={"next": request.get_full_path()},
+                ),
+                iconname="edit",
+                label=_("Edit"),
+            ),
+            Link(
+                href=ModelHref(
+                    Relationship,
+                    "delete",
+                    kwargs={"pk": hg.C("row.pk")},
+                    query={"next": request.get_full_path()},
+                ),
+                iconname="trash-can",
+                label=_("Delete"),
+            ),
+        ],
+        primary_button=primary_button,
+    )
+
+
+def button_add_relationship_from(modal):
+    return layout.button.Button(
+        _("Add relationship from person"),
+        buttontype="primary",
+        **modal.openerattributes,
+    )
+
+
+def button_add_relationship_to(modal):
+    return layout.button.Button(
+        _("Add relationship to person"),
+        buttontype="primary",
+        **modal.openerattributes,
+    )
+
+
+def modal_add_relationship_from(person):
+    ret = layout.modal.Modal.with_ajax_content(
+        heading=_("Add relationship from person"),
+        url=ModelHref(
+            Relationship, "add", query={"asajax": True, "person_a": person.pk}
+        ),
+        submitlabel=_("Save"),
+    )
+    return ret
+
+
+def modal_add_relationship_to(person):
+    ret = layout.modal.Modal.with_ajax_content(
+        heading=_("Add relationship to person"),
+        url=ModelHref(
+            Relationship, "add", query={"asajax": True, "person_b": person.pk}
+        ),
+        submitlabel=_("Save"),
+    )
+    return ret
+
+
+def person_in_relationship(
+    header: str,
+    field_name: str,
+    get_person: Callable[[Relationship], Person],
+) -> DataTableColumn:
+    return DataTableColumn(
+        header,
+        hg.SPAN(
+            person_name(field_name),
+            person_number_in_brackets(field_name),
+            **attributes_for_link_to_person(get_person),
```

Comment:
This one you might want to assign to the DataTableColumn.td_attributes.
I think the SPAN will not fill the whole cell by default.

Reply:
(I mean only taking the ```attributes_for_link_to_person(get_person)``` part out, the SPAN is not a problem)

Reply:
Whether you really want to change this depends of course on how you want the UI to behave. If the full-cell-click is not desired then there is no need to change it.

Reply:
I just reverted this from an old version where we had these `attributes_for_link_to_person`... I think frederik wants to redesign it anyway.
Currently I am only trying to get rid of the global submit button :)